### PR TITLE
#6 GET /api/users

### PR DESCRIPTION
**Pull request by ajknowlden1**
-creates the endpoint
-adds controllers/models
-tests the endpoint

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -88,6 +82,28 @@ describe("GET /api/articles/:article_id", () => {
...
       });
+  });
+});
 
+describe("GET /api/users", () => {
+  it("should return an array", () => {
+    return request(app)
+      .get("/api/users")
+      .expect(200)
+      .then((response) => {
```
Comment (mallionaire) [inline __tests__/app.test.js:96]: I think you might need to refactor this, your users should be sent back as an obj with a key of `users` so you might want to take another look at how you'er sending this back from your controller

File: __tests__/app.test.js [hunk 2]
```diff
@@ -88,6 +82,28 @@ describe("GET /api/articles/:article_id", () => {
...
+        expect(Array.isArray(response.body)).toBe(true);
+      });
+  });
+  it("should return an array of user objects with the correct keys", () => {
+    return request(app)
+      .get("/api/users")
+      .expect(200)
+      .then((response) => {
+        const userArr = response.body;
```
Comment (mallionaire) [inline __tests__/app.test.js]: I'd recommend trying to destructure the value from the response.body eg `const { users } = response.body` - it gives your tests an added bit of security, and makes sure we're sending back the users on the correct key 😄

File: controllers/users.controllers.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const res = require("express/lib/response");
+const { selectUsers } = require("../models/users.models");
+
+const getUsers = (req, res, next) => {
+  selectUsers()
+    .then((response) => {
+      res.status(200).send(response);
```
Comment (mallionaire) [inline controllers/users.controllers.js]: Look closely at what you're sending... what data type is is?

**Review by mallionaire — CHANGES_REQUESTED** [review]
There are a couple of files you've deleted, I assume this was a project restructure. Very good choice 👍

**Comment by ajknowlden1** [comment]
I've implemented the feedback you've suggested, let me know if there's anything I've missed!

**Pull request merged**